docs/src/tutorials/lowrank.md

```diff
@@ -21,42 +35,42 @@ Define lr-space dimensions
 N_cut = 2         # Number of states of each mode
 N_modes = latt.N  # Number of modes
 N = N_cut^N_modes # Total number of states
-M = Nx * Ny + 1       # Number of states in the LR basis
+M = latt.N + 1       # Number of states in the LR basis
 ```
 
 Define lr states. Take as initial state all spins up. All other N states are taken as those with miniman Hamming distance to the initial state.
 
 ```@example lowrank
-ϕ = Vector{QuantumObject{Vector{ComplexF64},KetQuantumObject}}(undef, M)
-ϕ[1] = kron(repeat([basis(2, 0)], N_modes)...)
+ϕ = Vector{QuantumObject{Vector{ComplexF64},KetQuantumObject,M-1}}(undef, M)
+ϕ[1] = kron(fill(basis(2, 1), N_modes)...)
 
-global i = 1
+i = 1
 for j in 1:N_modes
     global i += 1
-    i <= M && (ϕ[i] = mb(sp, j, latt) * ϕ[1])
+    i <= M && (ϕ[i] = SingleSiteOperator(sigmap(), j, latt) * ϕ[1])
 end
 for k in 1:N_modes-1
     for l in k+1:N_modes
         global i += 1
-        i <= M && (ϕ[i] = mb(sp, k, latt) * mb(sp, l, latt) * ϕ[1])
+        i <= M && (ϕ[i] = SingleSiteOperator(sigmap(), k, latt) * SingleSiteOperator(sigmap(), l, latt) * ϕ[1])
     end
 end
 for i in i+1:M
     ϕ[i] = QuantumObject(rand(ComplexF64, size(ϕ[1])[1]), dims = ϕ[1].dims)
     normalize!(ϕ[i])
 end
+nothing # hide
 ```
 
 Define the initial state
 
 ```@example lowrank
-z = hcat(broadcast(x -> x.data, ϕ)...)
-p0 = 0.0 # Population of the lr states other than the initial state
-B = Matrix(Diagonal([1 + 0im; p0 * ones(M - 1)]))
+z = hcat(get_data.(ϕ)...)
+B = Matrix(Diagonal([1 + 0im; zeros(M - 1)]))
 S = z' * z # Overlap matrix
 B = B / tr(S * B) # Normalize B
 
-ρ = QuantumObject(z * B * z', dims = ones(Int, N_modes) * N_cut); # Full density matrix
+ρ = QuantumObject(z * B * z', dims = ntuple(i->N_cut, Val(N_modes))); # Full density matrix
```

Comment:
I did a benchmark.
The following method for generating tuple-type `dims` is faster and less memory allocation:
```julia
Tuple(ones(Int, N_modes) * N_cut)
```